src/test/sha256_test.cpp

```diff
@@ -0,0 +1,207 @@
+#include "gtest/gtest.h"
```

Comment:
Why is this test showing up in this PR?

Reply:
When doing tests on blake2s I saw that the sha256 were not correctly implemented or called. I took the opportunity to fix that (even if we now are not using sha256 anymore (irony)).

Reply:
> When doing tests on blake2s I saw that the sha256 were not correctly implemented or called

What was wrong with the sha256 gadget? Do you remember the changes you made? The diffs only show the full file because of the rebase I assume

Reply:
The gadget was fine. It was the test file which had wrong values (and if I remember well some problems with pointers).

Reply:
What kind of problems with pointers? 
So the test values were wrong, but the tests were passing?

Reply:
I have just checked and I do not  see any modification between this version and the one in develop. I do not really understand.

What I recall is that the test was not called and I had to make a few modifications. 

There was a problem with the values and I thought that was due to their initialization, the pointers or the init_public_param. I cannot recall.

Reply:
We'll need to investigate that because we can't merge it without understanding why the file has been changed and what modifs were made (let's follow a "1 PR-1 thing" approach so that we know exactly what is merged)

Reply:
Should I overwrite this doc with develop's version then?

Reply:
Let's find out what changes you've done (you can compare the files on this branch and on develop to figure this out I guess) and move this into a separate PR as this is not related to blake2s

Reply:
This shows up because I wanted to rename test_sha256 into sha256_test as all the tests are written in that fashion ("xxx_test").

The current test_sha256 compilation fails, because there is no annotation given during the variable allocation and the interface of HashT must have somehow changed  since we have "libsnark::SHA256_block_size" as an input of the constructor in the tests (which is wrong).

The modifications I made includes changing the name (it created a new file and should have deleted test_sha256 but I can still see it), removing this sha256_block_size, adding the annotations and I am not sure why replacing pointer variables by variables.

Reply:
I'll open a new PR to enforce these changes (not the pointers part) as said previously.

Reply:
Issue created, and changes reverted.

Reply:
Please make sure to point to the issue for the record. ie: https://github.com/clearmatics/zeth/issues/83